packages/examples/packages/interactive-ui/src/index.ts

```diff
@@ -0,0 +1,331 @@
+import { rpcErrors } from '@metamask/rpc-errors';
+import type {
+  OnHomePageHandler,
+  OnTransactionHandler,
+  OnUserInputHandler,
+  Transaction,
+  Component,
+} from '@metamask/snaps-sdk';
+import {
+  panel,
+  type OnRpcRequestHandler,
+  heading,
+  button,
+  UserInputEventType,
+  form,
+  input,
+  ButtonType,
+  text,
+  copyable,
+  ManageStateOperation,
+  assert,
+  address,
+  row,
+} from '@metamask/snaps-sdk';
+import { hasProperty } from '@metamask/utils';
+
+import { decodeData } from './utils';
+
+/**
+ * Initiate a new interface with the starting screen.
+ *
+ * @returns The Snap interface ID.
+ */
+export async function createInterface(): Promise<string> {
+  return await snap.request({
+    method: 'snap_createInterface',
+    params: {
+      ui: panel([
+        heading('Interactive UI Example Snap'),
+        button({ value: 'Update UI', name: 'update' }),
+      ]),
+    },
+  });
+}
+
+/**
+ * Create the transaction insights components to display.
+ *
+ * @returns The transaction insight content.
+ */
+export async function getInsightContent(): Promise<Component> {
+  const snapState = await snap.request({
+    method: 'snap_manageState',
+    params: {
+      operation: ManageStateOperation.GetState,
+    },
+  });
+
+  assert(snapState?.transaction, 'No transaction found in Snap state.');
+
+  const { from, to } = snapState.transaction as Transaction;
+
+  return panel([
+    row('From', address(from)),
+    row('To', to ? address(to) : text('None')),
+    button({ value: 'See transaction type', name: 'transaction-type' }),
+  ]);
+}
+
+/**
+ * Updates a Snap interface to display the transaction type.
+ *
+ * @param id -  The interface ID to update.
+ */
+export async function displayTransactionType(id: string) {
+  let type = 'unkown';
```

Comment:
Maybe change the `decodeData` function to support `undefined` instead of using mutable variables.